modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/ballooneditor/plugin.js

```diff
@@ -180,26 +196,38 @@
 					x =
 						selectedElementClientRect.x +
 						selectedElementClientRect.width / 2 -
-						panelClientRect.width / 2;
+						panelRect.width / 2;
 					y =
 						selectedElementClientRect.y -
-						panelClientRect.height -
+						panelRect.height -
 						padding;
 				}
 
-				if (x < editableClientRect.x) {
-					x = editableClientRect.x + padding;
+				const editable = this.editor.editable();
+
+				const editableRect = editable.getClientRect(true);
+
+				if (editableRect.width < panelRect.width + padding) {
+					x = editableRect.x + padding;
 				}
-				if (x + panelClientRect.width > editableClientRect.width) {
-					x -= x + panelClientRect.width - editableClientRect.width;
+				else if (x < editableRect.x) {
```

Comment:
Did you changed these formulas? Or is just a weird diff?

Reply:
I changed them, in some cases significantly.